tests/test_examples.py

```diff
@@ -943,3 +945,32 @@ def test_example_get_current(self):
                 -1.36284423e07,
             ],
         )
+
+
+@pytest.mark.unit
+def test_dummy_for_optimizing_collections():
+    """Test with a dummy objective for optimizing collections."""
+    eq = Equilibrium()
+    tf = ToroidalMagneticField(1.0, 11.0)
+    pf = VerticalMagneticField(4.0)
+    field = tf + pf
+
+    # use grid with only points at R=11m (same as the TF field R0)
+    # so that the end result should be TF B0 / VF B0 = 2
+    objective = ObjectiveFunction(
+        DummyFields(
+            field,
+            eq,
+            target=2,
+            eval_grid=LinearGrid(rho=np.array(1.0), theta=np.array(0.0), N=1),
+        )
+    )
+    optimizer = Optimizer("lsq-exact")
+    constraints = (FixParameter(field, [["R0"], []]),)
```

Comment:
Currently I think the API I have setup (have not changed docs yet) is that if it is an optimizableCollection, you MUST pass in params that is a list of lists, the outer list being `len(OptimizableCollection)`, to avoid ambiguity in what is being input. Then if you say only want to fix one subthing's param, like in the case here, you leave the other list of params empty. 

This case is fixing only the `"R0"` of the Tf field but leaving its `"B0"` and the vertical field's `"B0"` free

Reply:
something like this might be easier? https://jax.readthedocs.io/en/latest/jax-101/05.1-pytrees.html#key-paths